include/matx/core/tie.h

```diff
@@ -0,0 +1,112 @@
+////////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2021, NVIDIA Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+
+#pragma once
+
+#include <tuple>
+#include "matx/core/type_utils.h"
+
+namespace matx {
+
+/* mtie exists because we can't overload std::tuple's operator= for operator assignments in
+   a multiple return set statement. */
+template <typename... Ts>
+struct mtie : public BaseOp<mtie<Ts...>>{
+  using mtie_type = bool;
+  using scalar_type = void; // Doesn't matter since it's not used
+  using value_type  = void; // Doesn't matter since it's not used
+  using shape_type  = index_t;
+  using matxoplvalue = bool;
+
+  static_assert(sizeof...(Ts) > 1, "Only use mtie() when number of elements is greater than 1");
+
+  __MATX_INLINE__ const std::string str() const {
+    if constexpr (sizeof...(Ts) == 2) {
+      return get_type_str(std::get<0>(ts_)) + "=" + get_type_str(std::get<1>(ts_));
+    }
+    else {
+      // Figure this out for proper LHS
```

Comment:
Is there more work here?  We we can't figure this out now can we add a static assert here to show that this is not working?

Reply:
I'll figure it out, but I was getting a compiler issue I wasn't sure if it was a bug. Basically for cases like `((U, S, V) = svd(A))` we want the LHS to print `(TF32, TF32, TF32)` to make it clear there are multiple return types